### DISCUSSION TITLE
Shouldn't com.apple.WebKit.Networking be recognised as bot?

### DISCUSSION THREAD

**Discussion by Marcdj-02** [omrilotan/isbot#260] [General]
I saw the following request come through when visiting `http://localhost:3000/login`:
`GET http://localhost:3000/apple-touch-icon.png` with the userAgent: `com.apple.WebKit.Networking/xxx.x.xx.xx.x Network/xxxx.x.xx.xxx.x macOS/15.0`.

It seems to me this request is a background process, I am not sure if this userAgent pattern is only used for background processes, if so shouldn't it be added to the bot patterns?

Thanks in advance for the replies!

**Comment by omrilotan** [comment]
Thank you. I can see your point and I agree with the logic. I'm not 100% sure I can define background processes related to user sessions as "bot", but I can see how this can be considered as an automated request.
I will try finding data to back it up, and then release an update to the recognition pattern.

**Reply by Marcdj-02** [reply]
Thanks for your quick reply! I am a aware this might be a weird case, but worth taking a look at IMO

**Comment by omrilotan** [comment]
Looks like it's quite the opposite. I wouldn't call this process automated. It's just a background process for actual user apps. What do you think, @Marcdj-02 ?

https://developer.apple.com/documentation/browserenginekit/networkingextension

> An interface for configuring a networking helper extension process that will carry out networking operations on behalf of the browser app

**Reply by Marcdj-02** [reply]
If I read this correctly, this is used for browser extensions. Normally, as far as I know, extension like Chrome use the actual user-agent of the browser it is ran in, along with the cookies of the user.

Unfortunately, I am not able to reproduce this, yet I am quite certain the user's cookies weren't sent along, which makes sense if the user-agent is different. This indicated that is a 'different' process and therefore I'd see it as an automated process. If this makes it a bot or not is quite tricky.

For your context, I am using `isbot` for tracking unauthenticated visitors, whilst excluding bots, and such a request would increase the visitors, which it shouldn't. But it sound more like a me-problem, than that it is missing from the bot quantification.

Thanks anyways!

**Reply by omrilotan** [reply]
You're right `com.apple.WebKit.Networking` is not used by browser extensions but the definition is very similar.

It makes sense that the users' cookies will not be included in these requests.

Still doesn't make it a bot request necessarily, and definitely not a self-identifying bot.

I am going to close this issue but thank you for your attention.
